usdt-impl/src/common.rs

```diff
@@ -7,11 +7,18 @@ use quote::{format_ident, quote};
 // Construct function call that is used internally in the UDST-generated macros, to allow
 // compile-time type checking of the lambda arguments.
 pub fn generate_type_check(types: &[dtrace_parser::DataType]) -> TokenStream {
+    let mut has_strings = false;
```

Comment:
I might return a tuple of `(TokenStream, bool)` from the match statement, rather than mutating this, but not crucial.

Reply:
Nvm, I see now that this is inside a map. We'd have to reduce this outside, which might not be any easier or clearer. Disregard :)

Reply:
So that would l look like: `type_check_args.iter().map(..).unzip()` and then `out.1.iter().reduce(|a, b|a || b)` I think? ... and that seemed a little less clear. What do you think?

Reply:
Yeah, definitely no better, and probably a good bit worse :)